packages/node/example/multiSession/src/serverSideApp.ts

```diff
@@ -0,0 +1,164 @@
+
+// The only import we need from the Node AuthN library is the Session class.
+import {
+  getSessionFromStorage,
+  Session,
+  getSessionIdFromStorageAll,
+} from "../../../dist/Session";
+
+import cookieSession from "cookie-session";
+
+const clientApplicationName = "solid-client-authn-node multi session demo";
+
+import express from "express";
+const app = express();
+const PORT = 3001;
+
+const DEFAULT_OIDC_ISSUER = "https://broker.pod.inrupt.com/";
+
+app.use(
+  cookieSession({
+    name: "session",
+    // These keys are required by cookie-session to sign the cookies.
+    keys: ["key1", "key2"],
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours
+  })
+);
+
+app.get("/", async (req, res, next) => {
+  const sessionIds = await getSessionIdFromStorageAll();
+  const sessions = await Promise.all(
+    sessionIds.map(async (sessionId) => {
+      return await getSessionFromStorage(sessionId);
+    })
+  );
+  const htmlSessions =
+    sessions.reduce((sessionList, session) => {
+      if (session?.info.isLoggedIn) {
+        return sessionList + `<li>${session?.info.webId}</li>`;
+      }
+      return "Anonymous";
+    }, "<ul>") + "</ul>";
+  res
+    .writeHead(200, { "Content-Type": "text/html" })
+    .write(
+      `<p>There are currently ${sessionIds.length} visitors: ${htmlSessions}</p>`
+    );
+  res.end();
+});
+
+app.get("/login", async (req, res, next) => {
+  const session = new Session();
+  req.session!.sessionId = session.info.sessionId;
+  await session.login({
+    redirectUrl: "http://localhost:3001/redirect",
+    oidcIssuer: DEFAULT_OIDC_ISSUER,
+    clientName: clientApplicationName,
+    handleRedirect: (data) => {
+      res.writeHead(302, {
+        location: data,
+      });
+      res.end();
+    },
+  });
+  if (session.info.isLoggedIn) {
+    res
+      .writeHead(200, { "Content-Type": "text/html" })
+      .write(`<p>Already logged in with WebID: [${session.info.webId}]</p>`);
+    res.end();
+  }
+});
+
+app.get("/redirect", async (req, res) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (session === undefined) {
+    res
+      .writeHead(400, { "Content-Type": "text/html" })
+      .write(`<p>No session stored for ID ${req.session!.sessionId}</p>`);
+  } else {
+    await session.handleIncomingRedirect(getRequestFullUrl(req));
+    if (session.info.isLoggedIn) {
+      res
+        .writeHead(200, { "Content-Type": "text/html" })
+        .write(`<p>Logged in as [${session.info.webId}] after redirect</p>`);
+    } else {
+      res
+        .writeHead(400, { "Content-Type": "text/html" })
+        .write(`<p>Not logged in after redirect</p>`);
+    }
+  }
+  res.end();
+});
+
+app.get("/fetch", async (req, res, next) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (!req.query["resource"]) {
+    res
+      .writeHead(400, { "Content-Type": "text/html" })
+      .write("<p>Expected a 'resource' query param</p>");
+  } else if (session) {
```

Comment:
Feels like everything after the `if` can be compressed with a `const fetch = (session ?? new Session()).fetch`.

Reply:
Indeed 6168832e4ece408a454a56b0c83eae428491240d